Index: LF7_v1.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>import RPi.GPIO as GPIO\r\nimport dht11\r\nimport board\r\nimport time\r\nimport busio\r\nimport os\r\nimport sys\r\n\r\nfrom adafruit_ht16k33.segments import Seg7x4\r\n\r\n# initialize GPIO\r\nGPIO.setwarnings(False)\r\nGPIO.setmode(GPIO.BCM)\r\nGPIO.cleanup()\r\ninstance = dht11.DHT11(pin = 4)\r\nresult = instance.read()\r\n\r\nwhile not result.is_valid():\r\n\tresult = instance.read()\r\nprint(\"Temperature: %-3.1f C\" % result.temperature)\r\nprint(\"Humidity: %-3.1f %%\" % result.humidity)\r\ntime.sleep(15)\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/LF7_v1.py b/LF7_v1.py
--- a/LF7_v1.py	(revision 6862af1c790e4a152ad0193265fd26df2a55eb22)
+++ b/LF7_v1.py	(date 1671108204413)
@@ -16,7 +16,9 @@
 result = instance.read()
 
 while not result.is_valid():
-	result = instance.read()
+	if result.is_valid:
+		result = instance.read()
+
 print("Temperature: %-3.1f C" % result.temperature)
 print("Humidity: %-3.1f %%" % result.humidity)
 time.sleep(15)
